examples/middleware/e2e/base.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { test, expect } from "@playwright/test";
+
+test("redirect", async ({ page }) => {
+  await page.goto("/");
+  await page.click('[href="/about"]');
+  expect(page.waitForURL("**/redirected"));
+  expect(await page.textContent("h1")).toContain("Redirected");
+});
+
+test("rewrite", async ({ page }) => {
+  await page.goto("/");
+  await page.click('[href="/another"]');
+  expect(page.waitForURL("**/another"));
+  expect(await page.textContent("h1")).toContain("Rewrite");
+});
+
+test("noop middleware", async ({ page }) => {
+  await page.goto("/");
+  await page.click('[href="/about2"]');
```

Comment:
My assumption based on the matchers is that this shouldn't match the middleware function - is it worth adding one that also matches the NextResponse.next() part too?

Reply:
> My assumption based on the matchers is that this shouldn't match the middleware function

That's a correct assumption. This test is here to verify that assumption.

> is it worth adding one that also matches the NextResponse.next() part too?

That should be the previous test (/another)

I think this can be marked as resolved unless I miss something

Reply:
`/another` returns the rewrite so i don't think it would cover the `NextResponse.next()` bit. I can't remember if rewrites would invoke middleware again because it's been a while since i last did a rewrite in a middleware, but the rewritten path doesn't match the matcher so even if it did, i dont think it's covered

Reply:
I double checked middleware.ts and you are absolutely right. We could add a /passthrough route and test it.

Reply:
added a /middleware route + test